Sources/xcodeproj/Objects/Project/PBXProject.swift

```diff
@@ -191,6 +191,44 @@ public final class PBXProject: PBXObject {
 
         return reference
     }
+    
+    
+    /// Adds a local swift package
+    ///
+    /// - Parameter path: Relative path to the swift package
+    /// - Parameter productName: The product to depend on
+    /// - Parameter target: Target to link package product to
+    /// - Parameter addFileReference: Include a file reference to the package (defaults to main group)
+    public func addLocalSwiftPackage(path: String,
+                                     productName: String,
+                                     target: PBXTarget? = nil,
+                                     addFileReference: Bool = true) -> XCSwiftPackageProductDependency {
+        let objects = try! self.objects()
+        
+        // Product
+        let productDependency = XCSwiftPackageProductDependency(productName: productName)
+        objects.add(object: productDependency)
+        target?.packageProductDependencies.append(productDependency)
+
+        // Build file
+        let buildFile = PBXBuildFile(product: productDependency)
+        objects.add(object: buildFile)
+        
+        // Link the product
+        try? target?.frameworksBuildPhase()?.files?.append(buildFile)
+        
+        // File reference
+        if addFileReference {
+            let reference = PBXFileReference(sourceTree: .group,
+                                             name: productName,
+                                             lastKnownFileType: "folder",
```

Comment:
Is this necessary?

Reply:
Do you mean `addFileReference` boolean? If so, that's something we (along with @kwridan ) wanted to take care of in Tuist - local packages' references are added to the top of the children in the main group, so in `tuist` we wanted to achieve consistency with that and that was not possible without adding the reference ourselves before calling this method.

Reply:
I see what you mean. What about adding a comment in the line `220` explaining this scenario? I think it's not obvious why someone might want to add a Swift package without adding the reference.